pkg/kubelet/kubelet.go

```diff
@@ -1024,42 +1017,14 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 	return node, nil
 }
 
-// getNodeLabels is just a wrapper method for the two below, not to duplicate above
+// getNodeLabels extracts the node labels specified on the command line
 func (kl *Kubelet) getNodeLabels() (map[string]string, error) {
-	var err error
-	labels := make(map[string]string, 0)
-
-	if kl.nodeLabelsFile != "" {
-		labels, err = kl.retrieveNodeLabelsFile(kl.nodeLabelsFile)
-		if err != nil {
-			return labels, err
-		}
-	}
-	// step: apply the command line label - permitted to override those from file
-	if len(kl.nodeLabels) > 0 {
-		nl, err := kl.retrieveNodeLabels(kl.nodeLabels)
-		if err != nil {
-			return labels, err
-		}
-		for k, v := range nl {
-			if vl, found := labels[k]; found {
-				glog.Warningf("the --node-label %s=%s option will overwrite %s from node-labels-file", k, v, vl)
-			}
-			labels[k] = v
-		}
-	}
-
-	return labels, nil
-}
-
-// retrieveNodeLabels extracts the node labels specified on the command line
-func (kl *Kubelet) retrieveNodeLabels(labels []string) (map[string]string, error) {
 	nodeLabels := make(map[string]string, 0)
 
-	for _, label := range labels {
-		items := strings.Split(label, "=")
+	for _, label := range kl.nodeLabels {
+		items := strings.Split(label, ":")
```

Comment:
Is there any precedent for having json / yaml supplied as a flag in our binaries?  I don't have a strong feeling about this, but do think that giving json as cli flags is less intuitive and visually appealing.  Since we plan to migrate these to a config soon anyway, that may not matter much.  If you still prefer using json/yaml I will make the changes.

Pros of Unmarshaling:
Can reuse same code for flags and config lables

Cons of Unmarshaling:
Makes flag harder to specify by adding boiler plate
Lose the multiple flag declarations semantics